### DISCUSSION TITLE
Issue with playing Ant Media Server VODs on MacOS after recording on Windows OS.

### DISCUSSION THREAD

**Discussion by asharamseervi** [ant-media/Ant-Media-Server#4988] [FAQs]
With Ant Media Server after recording the stream on a Windows machine using API, the VOD plays fine on Windows. 

But when I play the same VOD on macOS using Quick Time Player v10.5, the video freezes after some seconds while the audio continues to work.

Worth to mention, VODs playback with Quick Time Player is fine for recordings made on macOS.

How can I overcome this?
Or, is it an expected behavior?

**Comment by Mohit-3196** [comment]
Thank you for this question.

This is a known issue in the quick-time player. This problem also exists for MacOS/iOS and Safari. Let me tell you the cause of the problem and offer a solution.

**Problem:**
The resolution may be changed in WebRTC sessions according to the network conditions so that the resolution of the recording is being changed to a lower or higher resolution.

Most of the players and browsers can handle that. On the other hand, Safari and Quick Time Player cannot handle resolution changes and the problem you mention appear.

**Solution:**
Transcoding the stream into a specific resolution with `ffmpeg` or using adaptive bitrate on the server side resolves this issue.
A typical ffmpeg command like the below one is sufficient
`ffmpeg -i INPUT.mp4 OUTPUT.mp4`

Thank you.
